Ref Arch Intro (#809)
gruntwork-io/docs: _docs-sources/refarch/whats-this/understanding-the-deployment-process.md

# Understanding the Deployment Process

The Gruntwork Reference Architecture has three deployment phases.

### 1. Configuration

Configuration of the Gruntwork Reference Architecture is primarily the customer's responsibility.

- We deliver a templated `infrastructure-live-${COMPANY_NAME}` repository to you in our GitHub organization
- You access the repo in GitHub via invitation in the [Gruntwork Dev Portal](https://app.gruntwork.io)
- You use the Gruntwork CLI wizard to create accounts and set config options
- Pre-flight checks run via Github Actions to determine when the repo is ready for deployment

> - Your accounts should be empty at conclusion of this phase

Reviewer: ```suggestion
- The AWS accounts you are deploying the Reference Architecture to should be empty at conclusion of this phase
```

- You merge the PR to the `main` branch to initiate the deployment phase

### 2. Deployment

The deployment phase is primarily Gruntwork's responsibility.

- We monitor the deployment and fix any errors that occur as needed

> - In some cases, back-and-forth with you may be required to resolve issues (e.g. AWS quota problems)

Reviewer: ```suggestion
- In some cases, we may need to communicate with you to resolve issues (e.g. AWS quota problems)
```

- Deployment is completed and the `infrastructure-live-${COMPANY_NAME}` repo is populated

> - You should not touch accounts during this phase

Reviewer: ```suggestion
- During the deployment phase, you should not attempt to modify resources in or respond to any automated notifications from your AWS accounts
```

> - You receive an email notification when the deployment is complete

Reviewer: ```suggestion
- Once the deployment is complete, you will receive an email
```

### 3. Adoption

The adoption phase is primarily the customer's responsibility.

- You complete “last mile” configuration following our handoff docs, including final Pipelines integrations with your CI/CD of choice
- You migrate the `infrastructure-live-${COMPANY_NAME}`repo to your own Version Control System or Github Organization
- You revoke Gruntwork access to your AWS account

> - Your accounts are once again fully in your control

Reviewer: ```suggestion
- At this points, your AWS accounts are fully in your control
```
